hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -1979,5 +2004,17 @@
       milestone: V2
       tests: []
     }
+    {
```

Comment:
Does SW do all this when servicing the NMI interrupt? If yes, then it sounds more like a system level usecase than hardware DV. If yes, then it should be moved to a separate `chip_sw_usecases_testplan.hjson` which should capture all those things in one place.

Reply:
@cfrantz could either of you help provide a little more context for this use-case?
I'm not completely sure what the SW role in this would be.

Reply:
I don't recall discussing this case, but this does sound similar to something the mask ROM will want to do during a fault event - namely triggering an SRAM initialization event.

In this test case, it seems like we're expecting an NMI to be raised and for software to then reinitialize SRAM to wipe it.  In the mask ROM's shutdown code, I'm planning on setting `SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT` in the `SRAM_CTRL_CTRL_REG_OFFSET` register to cause the re-initialization.

Reply:
hmm, got it,  so we will just request a new key rather than set `SRAM_CTRL_CTRL_INIT` as well?

Reply:
For testing, I think we should test both.

The reason I chose to request a new key for mask ROM shutdown is that access to SRAM is blocked during the key renewal and the rest of the ROM shutdown process can continue without touching SRAM.

Reply:
got it, thanks!
The testcase as written here has SW writing both CTRL bits, so we can just leave it as is.

@sriyerg I'd placed this test in the "System-level Scenarios" section, should we spin this off into a completely separate testplan?

Reply:
Keep it there for now - i will spin those out to a different file later.